security/pkg/server/ca/server.go

```diff
@@ -271,7 +271,11 @@ func (s *Server) applyServerCertificate() (*tls.Certificate, error) {
 func (s *Server) authenticate(ctx context.Context) *authenticate.Caller {
 	// TODO: apply different authenticators in specific order / according to configuration.
 	for _, authn := range s.authenticators {
-		if u, err := authn.Authenticate(ctx); u != nil && err == nil {
+		u, err := authn.Authenticate(ctx)
+		if err != nil {
+			log.Warnf("Authentication failed: %v", err)
```

Comment:
Suggestion to change to "Authentication failed with auth source %v..." to be more consistent with below message.

Reply:
if fail, current code always return caller as nil..